catalog/CatalogDatabaseCache.cpp

```diff
@@ -61,15 +61,17 @@ void CatalogDatabaseCache::update(const serialization::CatalogDatabase &proto) {
 
   SpinSharedMutexExclusiveLock<false> lock(relations_mutex_);
   for (int i = 0; i < proto.relations_size(); ++i) {
-    auto relation_schema = make_unique<const CatalogRelationSchema>(proto.relations(i));
-    const relation_id rel_id = relation_schema->getID();
+    const serialization::CatalogRelationSchema &proto_relation = proto.relations(i);
+    const relation_id rel_id = proto_relation.relation_id();
```

Comment:
@hbdeshmukh Ah, now I got that you were talking about the big `ExclusiveLock`. I wish you could comment on Line 62 regarding this concern.

Actually, I could refactor it in this PR. Thanks!